src-java/connecteddevices-topology/connecteddevices-storm-topology/src/main/java/org/openkilda/wfm/topology/connecteddevices/service/PacketService.java

```diff
@@ -94,28 +103,60 @@ public void handleLldpData(LldpInfoData data) {
         });
     }
 
-    private FlowRelatedData findFlowRelatedData(LldpInfoData data) {
-        if (data.getCookie() == LLDP_POST_INGRESS_COOKIE) {
+    /**
+     * Handle Arp info data.
+     */
+    public void handleArpData(ArpInfoData data) {
+        transactionManager.doInTransaction(() -> {
+
+            FlowRelatedData flowRelatedData = findFlowRelatedData(data);
+            if (flowRelatedData == null) {
+                return;
+            }
+
+            SwitchConnectedDevice device = getOrBuildArpDevice(data, flowRelatedData.originalVlan);
+
+            if (device == null) {
+                return;
+            }
+
+            device.setTimeLastSeen(Instant.ofEpochMilli(data.getTimestamp()));
+            device.setFlowId(flowRelatedData.flowId);
+            device.setSource(flowRelatedData.source);
+
+            switchConnectedDeviceRepository.createOrUpdate(device);
+        });
+    }
+
+    private FlowRelatedData findFlowRelatedData(ConnectedDevicePacketBase data) {
+        long cookie = data.getCookie();
+        if (cookie == LLDP_POST_INGRESS_COOKIE
+                || cookie == ARP_POST_INGRESS_COOKIE) {
             return findFlowRelatedDataForVlanFlow(data);
-        } else if (data.getCookie() == LLDP_POST_INGRESS_VXLAN_COOKIE) {
+        } else if (cookie == LLDP_POST_INGRESS_VXLAN_COOKIE
+                || cookie == ARP_POST_INGRESS_VXLAN_COOKIE) {
             return findFlowRelatedDataForVxlanFlow(data);
-        } else if (data.getCookie() == LLDP_POST_INGRESS_ONE_SWITCH_COOKIE) {
+        } else if (cookie == LLDP_POST_INGRESS_ONE_SWITCH_COOKIE
+                || cookie == ARP_POST_INGRESS_ONE_SWITCH_COOKIE) {
             return findFlowRelatedDataForOneSwitchFlow(data);
-        } else if (data.getCookie() == LLDP_INPUT_PRE_DROP_COOKIE
-                || data.getCookie() == LLDP_INGRESS_COOKIE
-                || data.getCookie() == LLDP_TRANSIT_COOKIE) {
+        } else if (cookie == LLDP_INPUT_PRE_DROP_COOKIE
+                || cookie == LLDP_INGRESS_COOKIE
+                || cookie == LLDP_TRANSIT_COOKIE
+                || cookie == ARP_INPUT_PRE_DROP_COOKIE
+                || cookie == ARP_INGRESS_COOKIE
+                || cookie == ARP_TRANSIT_COOKIE) {
             int vlan = data.getVlans().isEmpty() ? 0 : data.getVlans().get(0);
             return new FlowRelatedData(vlan, null, null);
         }
-        log.warn("Got LLDP packet from unknown rule with cookie {}. Switch {}, port {}, vlans {}",
+        log.warn("Got LLDP or ARP packet from unknown rule with cookie {}. Switch {}, port {}, vlans {}",
```

Comment:
Can we figure out which packet here, LLDP or ARP? If we can, then it would be better to log more accurate information.